iceoryx_posh/test/moduletests/test_popo_unique_port_id.cpp

```diff
@@ -26,160 +28,155 @@ using namespace ::testing;
+    UniquePortId a;
     auto id = static_cast<uint64_t>(a);
     decltype(a) b(std::move(a));
     EXPECT_THAT(static_cast<uint64_t>(b), Eq(id));
 }
 
-TYPED_TEST(TypedUniqueId_test, MoveAssignmentSetsSameID)
+TEST(UniquePortId_test, MoveAssignmentSetsSameID)
 {
     ::testing::Test::RecordProperty("TEST_ID", "5d8a8771-be7a-45d2-b2bb-be89938241b6");
-    typename TestFixture::UniqueIDType a, b;
+    UniquePortId a, b;
     auto id = static_cast<uint64_t>(a);
     b = std::move(a);
     EXPECT_THAT(static_cast<uint64_t>(b), Eq(id));
 }
 
-TYPED_TEST(TypedUniqueId_test, SameIDsAreEqual)
+TEST(UniquePortId_test, SameIDsAreEqual)
 {
     ::testing::Test::RecordProperty("TEST_ID", "4040fe82-3220-402d-8618-b152f6c1042e");
-    typename TestFixture::UniqueIDType a, b(a);
+    UniquePortId a, b(a);
     EXPECT_TRUE(a == b);
     EXPECT_TRUE(a <= b);
     EXPECT_FALSE(a != b);
     EXPECT_TRUE(a >= b);
 }
 
-TYPED_TEST(TypedUniqueId_test, DifferentIDsAreNotEqual)
+TEST(UniquePortId_test, DifferentIDsAreNotEqual)
 {
     ::testing::Test::RecordProperty("TEST_ID", "ca8c7eae-d2af-4560-9bb7-c2e876103a62");
-    typename TestFixture::UniqueIDType a, b;
+    UniquePortId a, b;
     EXPECT_FALSE(a == b);
     EXPECT_TRUE(a <= b);
     EXPECT_TRUE(a != b);
     EXPECT_FALSE(a >= b);
 }
 
-TYPED_TEST(TypedUniqueId_test, LatestIDIsGreatestID)
+TEST(UniquePortId_test, LatestIDIsGreatestID)
 {
     ::testing::Test::RecordProperty("TEST_ID", "1327e92a-bc07-4ec9-8bc5-ee5a935ccd66");
-    typename TestFixture::UniqueIDType a, b;
+    UniquePortId a, b;
     EXPECT_TRUE(a < b);
     EXPECT_TRUE(a <= b);
     EXPECT_FALSE(a > b);
     EXPECT_FALSE(a >= b);
 }
 
-TYPED_TEST(TypedUniqueId_test, FirstIDIsSmallestID)
+TEST(UniquePortId_test, FirstIDIsSmallestID)
 {
     ::testing::Test::RecordProperty("TEST_ID", "6951e91f-0112-4c97-b972-34ddeada4191");
-    typename TestFixture::UniqueIDType a, b;
+    UniquePortId a, b;
     EXPECT_FALSE(b < a);
     EXPECT_FALSE(b <= a);
     EXPECT_TRUE(b > a);
     EXPECT_TRUE(b >= a);
 }
 
-TYPED_TEST(TypedUniqueId_test, ConversionToUint64)
+TEST(UniquePortId_test, ConversionToUint64)
 {
     ::testing::Test::RecordProperty("TEST_ID", "c1c58736-9755-4736-b163-3c8cc26db80d");
-    typename TestFixture::UniqueIDType a, b;
+    UniquePortId a, b;
     uint64_t id = static_cast<uint64_t>(a);
     b = a;
     EXPECT_EQ(id, static_cast<uint64_t>(b));
 }
 
-TYPED_TEST(TypedUniqueId_test, CreatingAnUniqueIdWithDefaultCTorIsValid)
+TEST(UniquePortId_test, CreatingAnUniqueIdWithDefaultCTorIsValid)
 {
     ::testing::Test::RecordProperty("TEST_ID", "0d810ee1-8ddd-48b3-8b53-d4430dd4bbe6");
-    typename TestFixture::UniqueIDType a;
+    UniquePortId a;
     EXPECT_TRUE(a.isValid());
 }
 
-TYPED_TEST(TypedUniqueId_test, InvalidIdIsInvalid)
+TEST(UniquePortId_test, InvalidIdIsInvalid)
 {
     ::testing::Test::RecordProperty("TEST_ID", "d0576b8d-65d2-4b53-88c0-05078f434a41");
-    typename TestFixture::UniqueIDType a(InvalidId);
```

Comment:
Maybe we should test that:
* Moving valid id to invalid id makes the destination valid
* Moving invalid id to valid id makes the destination invalid
* Moving id makes the source always invalid

For the move constructor and assignment operator.

Reply:
Don't we already have this tests in the `NewType` and wouldn't this somewhat negate the benefits of the pattern. After all, it is a wrapper around a POD which does not have an undefined state after move.